cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java

```diff
@@ -0,0 +1,157 @@
+import com.amazonaws.services.schemaregistry.serializers.GlueSchemaRegistrySerializationFacade;
+import com.amazonaws.services.schemaregistry.serializers.GlueSchemaRegistrySerializerImpl;
+import com.amazonaws.services.schemaregistry.utils.AWSSchemaRegistryConstants;
+
+import kotlinx.serialization.SerializationException;
+import lombok.Getter;
+
+import org.apache.kafka.connect.data.*;
+import org.apache.kafka.connect.errors.DataException;
+import org.apache.kafka.connect.storage.Converter;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+public class CrossRegionReplicationMM2Converter implements Converter {
+    @Getter
+    private GlueSchemaRegistryDeserializationFacade deserializationFacade;
+    @Getter
+    private GlueSchemaRegistrySerializationFacade serializationFacade;
+    @Getter
+    private AwsCredentialsProvider credentialsProvider;
+    @Getter
+    private GlueSchemaRegistryDeserializerImpl deserializer;
+    @Getter
+    private GlueSchemaRegistrySerializerImpl serializer;
+    @Getter
+    private boolean isKey;
+
+    /**
+     * Constructor used by Kafka Connect user.
+     */
+    public CrossRegionReplicationMM2Converter(){};
+
+    /**
+     * Constructor accepting AWSCredentialsProvider.
+     *
+     * @param credentialsProvider AWSCredentialsProvider instance.
+     */
+    public CrossRegionReplicationMM2Converter(
+            GlueSchemaRegistryDeserializationFacade deserializationFacade,
+            GlueSchemaRegistrySerializationFacade serializationFacade,
+            AwsCredentialsProvider credentialsProvider,
+            GlueSchemaRegistryDeserializerImpl deserializerImpl,
+            GlueSchemaRegistrySerializerImpl serializerImpl) {
+
+        this.deserializationFacade = deserializationFacade;
+        this.serializationFacade = serializationFacade;
+        this.credentialsProvider = credentialsProvider;
+        this.deserializer = deserializerImpl;
+        this.serializer = serializerImpl;
+    }
+
+    /**
+     * Configure the MM2 Schema Replication Converter.
+     * @param configs configuration elements for the converter
+     * @param isKey true if key, false otherwise
+     */
+    @Override
+    public void configure(Map<String, ?> configs, boolean isKey) {
+        this.isKey = isKey;
+        new CrossRegionReplicationMM2ConverterConfig(configs);
+
+
+        credentialsProvider = DefaultCredentialsProvider.builder().build();
+
+        // Put the source and target regions into configurations respectively
+        Map<String, Object> sourceConfigs = new HashMap<>(configs);
+        Map<String, Object> targetConfigs = new HashMap<>(configs);
+
+        sourceConfigs.put(AWSSchemaRegistryConstants.AWS_REGION, configs.get(AWSSchemaRegistryConstants.AWS_SOURCE_REGION));
```

Comment:
Have we handle the case where expected key-value pair does not exist in the config map?